### ISSUE TITLE
[@angular-eslint/template/i18n] Request: Only check for string or unknown typed attributes

### ISSUE THREAD

**Issue by Samuel-Therrien-Beslogic** [angular-eslint/angular-eslint#1565]
**Description and reproduction of the issue**

```JSON
{
  "rules": {
    "@angular-eslint/template/<rule>": "error"
  }
}
```

```html
<my-component cardType="info"></my-component>
```
Where `cardType` is
```ts
type CardTypes = 'default' | 'error' | 'warning' | 'info'
```

The rule could leverage typescript to only check for attributes of type `string`, `any`, `unknown` or any union containing those. Literal string unions should be omitted.
This would allow for many less false-positives with default configurations.

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `15.2.1` |
| `@angular-eslint/template-parser`         | `15.2.1` |
| `@typescript-eslint/parser`               | `5.62.0` |
| `ESLint`                                  | `8.52.0` |
| `node`                                    | `18.14.0` |

```sh

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/


Angular CLI: 16.1.8
Node: 18.14.0
Package Manager: npm 9.4.2
OS: win32 x64

Angular: 16.1.9
... common, compiler-cli, core, forms, language-service
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1601.8
@angular-devkit/build-angular   16.1.8
@angular-devkit/core            16.2.8
@angular-devkit/schematics      16.2.8
@angular/cdk                    16.1.8
@angular/cli                    16.1.8
@angular/material               16.1.8
@schematics/angular             16.2.8
ng-packagr                      16.1.0
rxjs                            7.8.1
typescript                      4.9.5
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by json-derulo** [comment]
I don't really understand what you are asking for. Do you want a new rule, and if yes, what problem should it solve? Or do you want the existing `@angular-eslint/template/i18n` rule to be adjusted, like the title suggests?

**Comment by Samuel-Therrien-Beslogic** [comment]
> I don't really understand what you are asking for. Do you want a new rule, and if yes, what problem should it solve? Or do you want the existing `@angular-eslint/template/i18n` rule to be adjusted, like the title suggests?

I forgot to substitute `<rule>` when I originally opened the issue. I'm asking for an adjustment of the existing rule to reduce false-positives by inferring that literal string typed inputs don't need to be translated.

**Comment by ahnpnl** [comment]
I was think similarly, only check on type `string`, `unknown`, `any` would reduce a lot false positives.
